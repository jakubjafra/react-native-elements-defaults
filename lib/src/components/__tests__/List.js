const testHelperUtils = require('../../testHelperUtils');

const utils = require('../../utils');
const Overwrite = utils.Overwrite;

const name = 'List',
    __oldElement__ = require('react-native-elements/src/list/List'),
    __element__ = require('../List'),
    noTheme = new Overwrite(
        name,
        Overwrite.TYPE.FUNCTIONAL_COMPONENT_ATTRIBUTES,
        Overwrite.SCOPE.DEFAULTS,
        {}
    ),
    basicTheme = new Overwrite(
        name,
        Overwrite.TYPE.FUNCTIONAL_COMPONENT_ATTRIBUTES,
        Overwrite.SCOPE.DEFAULTS,
        {
            containerStyle: {
                backgroundColor: 'red',
            }
        }
    );

describe(name, () => {
    testHelperUtils.defaultTests(name, __oldElement__, __element__, noTheme, basicTheme);
});
