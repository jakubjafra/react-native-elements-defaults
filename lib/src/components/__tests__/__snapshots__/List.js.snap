// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List with basic theme applied List renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;

exports[`List with basic theme applied OldList renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;

exports[`List with no theme applied List renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;

exports[`List with no theme applied OldList and List returns the same thing if no theme is applied 1`] = `"[2mCompared values have no visual difference.[22m"`;

exports[`List with no theme applied OldList renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;
